### ISSUE TITLE
Passing an object key as parameter for value binding breaks in 6.3.4

### ISSUE THREAD

**Issue by juho** [ManuelDeLeon/viewmodel#301]
Discovered something weird the other day, in 6.3.4 if you pass an object key to an input field as the `value` binding, it fails with a peculiar error message `Can't access "name" of "a"` (or whichever character you're typing). Whatever you type into the input will become the key name it tries to find in the VM. Works okay in 6.3.3.

```js
formData: {
  position: ""
}
```

```jade
input(type="text", name="requisition.position", autocomplete="off", autocorrect="off", autocapitalize="on", spellcheck="false", $b="value: formData.position")
```

**Comment by ManuelDeLeon** [comment]
I'm looking into this but Meteor 1.6 broke the package testing setup I had. Just wanted to let you know.

**Comment by juho** [comment]
Thanks Manuel. Sure, we're still on 1.5.x here as well.

**Comment by arggh** [comment]
`ViewModel 6.3.4` also breaks click events, if I bind it like so

```
<button {{b "click: myItem.doSomething"}}>Do something</button>
```

**Comment by juho** [comment]
We noticed `this.parent()` complaining about this in 6.3.4 as well:

```
viewmodel-checks.coffee:79 viewmodel.parent() doesn’t take any arguments. It just returns the single parent view model. See https://viewmodel.org/docs/viewmodels#parent for more information
```

Maybe rolling back the release is a good idea.

**Comment by juho** [comment]
@ManuelDeLeon Happy new year! Had a chance to look at this stuff yet?

**Comment by juho** [comment]
Would be great if you can take a look soon, we're having a ton of IE trouble thanks to this (can't prevent events from cancelling w/ buttons and links) and have to convert `a` and `button` tagsto `div`s with click events which obviously kills keyboard navig.

**Comment by arggh** [comment]
@juho I just noticed, that if the click handler has the parenthesis at the end, implying a function call, like so...

```
<button {{b "click: parent.doSomething()"}}>Do something</button>
```

...ViewModel doesn't complain about `parent doesn't take any arguments` and it just works. So that's easily fixed by adding the parenthesis.

I also tested calling a function from an object returned by helper and a function specified in a namespaced ViewModel share and they also work, even without the parenthesis.

However, the initial issue about nested value bindings does still seem valid with ViewModel `6.3.7`, though I've never used such a pattern myself, I just tested it now.

**Comment by juho** [comment]
Oh hey thanks. Nice catch!

**Comment by arggh** [comment]
...I just noticed also, that with this code:

```handlebars
// child template
<button {{b "click: parent.action()"}}>Do action</button>
```

```js
// parent vm
action(event) {
   event.stopPropagation(); // will throw Uncaught TypeError: Cannot read property 'stopPropagation' of undefined
}
```

...the click event will not get passed to the `action` function.

**Comment by ManuelDeLeon** [comment]
Fixed on v6.3.8

@arggh In your case you're sending undefined (no arguments) to the parent action. Don't use the parenthesis `<button {{b "click: parent.action"}}>Action</button>`

**Comment by juho** [comment]
Thanks Manuel!
